File: apps/web/src/types/health.ts
export interface ComponentStatus {
  status: 'connected' | 'disconnected' | 'unknown';
  message?: string;
  latency_ms?: number;
}

export interface SystemHealthResponse {
  status: 'healthy' | 'degraded' | 'unhealthy';
  service: string;
  timestamp: string;
  version: string;
  components: {
    api: ComponentStatus;
    database: ComponentStatus;
    redis: ComponentStatus;
  };
}

File: apps/api/app/schemas/health.py
from pydantic import BaseModel


class ComponentStatus(BaseModel):
    status: str  # connected | disconnected | unknown
    message: str | None = None
    latency_ms: float | None = None


class SystemHealthResponse(BaseModel):
    status: str  # healthy | degraded | unhealthy
    service: str
    timestamp: str
    version: str
    components: dict[str, ComponentStatus]
